### PR TITLE
J7srcz5l CRUD functionality for Routes

### PR DESCRIPTION
**Pull request by KundwaBruno**
### WHAT DOES THIS PR DO?
This PR adds **CRUD** operations to route `/route`
This PR adds **CRUD** operations to route `/busStop`
### DESCRIPTION OF TASK TO BE COMPLETED
The operator should be able to create,delete and update a route.
The operator should be able to view a paginated list of routes in the system.
The users should be able to view a route's details.
Appropriate error messages are displayed on unsuccessful attempts.
### HOW SHOULD THIS BE MANUALLY TESTED?
**Testing** `/api/route` 

- Make a `.env` file and fill it with information as provided in the `.ev_example` file.
- First get a token granted from POST signin request on route `/api/users/signin` which is token you will be using to make requests along with the header `Authorization : token` on all routes because all routes are protected to only administratos and operators and also a body with email and password credentials of a sample admin from the mocks folder.

**Testing using postman**

- Make a POST request to route `/api/route` along with data that involves keys of start , end and distance and their values to **create** a route
- Make a GET request to route `/api/route` to **read** all routes' details in the database
- Make a GET request to route `/api/route/{routeID}` where id is the unique id given to a specific route to get that specific route's details
- Make a PATCH request to route `/api/route/{routeID}`  along with data that involves keys of start , end and distance and their values to **update** the specific route with its unique ID
- Make a DELETE request to route `/api/route/{routeID}` to **delete** that route with the specific ID
- Make a POST on route `/api/route/{routeID}` along with body that contains a key of `busStopId` and its value to assign a bus stop with the id to the route.

**Testing** `/api/busStop`

**Testing using postman**

- Make a POST request to route `/api/busStop` along with data that involves keys of sector , cell and coordinates and their values to **create** a bus stop
- Make a GET request to route `/api/busStop` to **read** all bus stop's details in the database
- Make a GET request to route `/api/busStop/{busStopId}` where `{busStopId}` is the unique id given to a specific bus stop to get that specific bus stop's details
- Make a PATCH request to route `/api/busStop/{busStopId}`  along with data that involves keys of sector , cell and coordinates and their values to **update** the specific bus stop with its unique ID
- Make a DELETE request to route `/api/busStop/{busStopId}` to **delete** that busStop with the specific ID

### RELEVANT TRELLO CARD
ID : J7srcz5l

### CODE REVIEW

**Review by mansurissa — APPROVED** [review]
LGTM

**Review by GuevaraMarius — APPROVED** [review]

**Review by SIBONIYO — APPROVED** [review]

**Review by 41hulk — COMMENTED** [review]
Nice Job Bruno

**Review by samuel776 — APPROVED** [review]

File: src/controllers/busStop.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    return succesRes(res, 200, res.__('All busStops'), foundBusStop);
+  } catch (error) {
+    return errorRes(res, 500, res.__('Internal server error : ') + error.message);
+  }
+};
+
+const oneStop = async (req, res) => {
+  try {
+    const busStopId = req.params;
```
Comment (raymond42) [inline src/controllers/busStop.js]: does really all the req.params equals to busStopId? check on this, we should get the busStopId from req.params not equalizing them

File: src/controllers/busStop.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  } catch (error) {
+    return errorRes(res, 500, res.__('Internal server error : ') + error.message,
+    );
+  }
+};
+
+const updateBusStop = async (req, res) => {
+  try {
+    const { sector, cell } = req.body;
+    const busStopId = req.params;
```
Comment (raymond42) [inline src/controllers/busStop.js]: same as here, you should destructure the busStopId from req.params not equalizing them

File: src/controllers/busStop.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    return errorRes(res, 404, res.__('bus stop not found :('));
+  } catch (error) {
+    return errorRes(res, res.__('Internal server error : ') + error.message,);
+  }
+};
+
+const deleteBusStop = async (req, res) => {
+  try {
+    const busStopId = req.params;
```
Comment (raymond42) [inline src/controllers/busStop.js]: same thing here

**Review by raymond42 — CHANGES_REQUESTED** [review]
good job. please attend to the comments and revert to me

**Review by raymond42 — APPROVED** [review]
LGTM

**Pull request merged**